invenio_communities/models.py

```diff
@@ -153,686 +101,72 @@ class Community(db.Model):
-        """Get user communities specified in recid."""
-        rec = get_record(recid).legacy_create_recstruct()
-        prefix = "%s-" % (
-            cfg['COMMUNITIES_ID_PREFIX_PROVISIONAL']
-            if provisional else cfg['COMMUNITIES_ID_PREFIX'])
-
-        colls = rec.get('980', [])
-        usercomm = []
-        for c in colls:
-            try:
-                # We are only interested in subfield 'a'
-                code, val = c[0][0]
-                if code == 'a' and val.startswith(prefix):
-                    val = val[len(prefix):]
-                    u = cls.query.filter_by(id=val).first()
-                    if u:
-                        usercomm.append(u)
-            except IndexError:
-                pass
-        return usercomm
-
-    @classmethod
-    def filter_communities(cls, p, so):
-        """Search for communities.
-
-        Helper function which takes from database only those communities which
-        match search criteria. Uses parameter 'so' to set communities in the
-        correct order.
-
-        Parameter 'page' is introduced to restrict results and return only
-        slice of them for the current page. If page == 0 function will return
-        all communities that match the pattern.
-        """
-        query = cls.query
-        if p:
-            query = query.filter(db.or_(
-                cls.id.like("%" + p + "%"),
-                cls.title.like("%" + p + "%"),
-                cls.description.like("%" + p + "%"),
-            ))
-        if so in cfg['COMMUNITIES_SORTING_OPTIONS']:
-            order = so == 'title' and db.asc or db.desc
-            query = query.order_by(order(getattr(cls, so)))
-        else:
-            query = query.order_by(db.desc(cls.ranking))
-        return query
-
-    #
-    # Utility methods
-    #
-    def get_collection_name(self, provisional=False):
-        """Get a unique collection name identifier."""
-        if provisional:
-            return "%s-%s" % (
-                cfg['COMMUNITIES_ID_PREFIX_PROVISIONAL'],
-                self.id)
-        else:
-            return "%s-%s" % (
-                cfg['COMMUNITIES_ID_PREFIX'], self.id)
-
-    def get_title(self, provisional=False):
-        """Get collection title."""
-        if provisional:
-            return "Provisional: %s" % self.title
+    def add_community(self, record):
+        """Add community to a record."""
+        if '_communities' not in record:
+            record['_communities'] = [self.id, ]
         else:
-            return self.title
+            assert isinstance(record['_communities'], list)
+            record['_communities'].append(self.id)
 
-    def get_collection_dbquery(self, provisional=False):
-        """Get collection query."""
-        return "%s:%s" % self.get_query(provisional=provisional)
+    def accept_record(self, record):
+        """Accept a record for inclusion in the community."""
+        cr = InclusionRequest.query.filter_by(
```

Comment:
Current state should be fine.

About the one-pass deletion - I actually need to a get an object to determine whether it exists first, and only then delete (otherwise log and raise an exception).